### ISSUE TITLE
Confusion on logging levels

### ISSUE THREAD

**Issue by criscola** [operator-framework/operator-sdk#4771]
## Type of question

Open question

#### What did you do?
```
ctrl.SetLogger(zap.Level(zapcore.Level(3))))
```
in my Reconcile method:
```
logger.V(3).Info("Reconcile called")
```
#### What did you expect to see?

I expected to see "Reconcile called" in my output logs.

#### What did you see instead? Under which circumstances?

Nothing.

#### Environment

**Operator type:**

/language go

#### Additional context

If I read the docs for [zapcore.Level](https://pkg.go.dev/go.uber.org/zap/zapcore@v1.15.0#Level), it says "A Level is a logging priority. Higher levels are more important.", but here's what the [docs for the Logger interface](https://pkg.go.dev/github.com/go-logr/logr@v0.3.0#Logger.V) is saying "[..] higher verbosity level means a log message is less important", so it looks like a contradiction. 

Am I misunderstanding how the logging levels are supposed to work? To me, the intention of the Logger interface appears more sensed, but when I try to manually set the logging level of zap, the behavior is not what I expect. Sorry if this is the wrong place to ask, but as the operator-sdk automatically scaffolds zap and logr together I thought it would be appropriated to talk about it here.

How do I set the desired level of logging manually ([not through BindFlag](https://sdk.operatorframework.io/docs/building-operators/golang/references/logging/#default-zap-logger))?

**Comment by kensipe** [comment]
In the process of investigating.   While this issue remains here for tracking, this is really a kubebuilder / controller-runtime issue / decision.   We will get clarification which hopefully results in documentation that will be posted here.

**Comment by kensipe** [comment]
https://kubernetes.slack.com/archives/CAR30FCJZ/p1618859524142700

**Comment by estroz** [comment]
TL;DR: the level should be `zapcore.Level(-3)`.

Created a PR to help clarify this in controller-runtime: https://github.com/kubernetes-sigs/controller-runtime/pull/1485

**Comment by criscola** [comment]
Thank you @estroz and @kensipe, this line of godoc is going to save some head-scratching 😃.

**Comment by kensipe** [comment]
for those inclined to read... I completely agree :)   thanks for reporting and following up!
